### DISCUSSION TITLE
Do I need to drop the table to import a new schema if keys change?

### DISCUSSION THREAD

**Discussion by iodev** [prisma/prisma#11711] [Q&A]
```
⚠️  There might be data loss when applying the changes:

  • A unique constraint covering the columns `[id]` on the table `Attribute` will be added. If there are existing duplicate values, this will fail.


√ Do you want to ignore the warning(s)? ... yes
Error: db error: ERROR: cannot drop constraint Attribute_pkey on table "Attribute" because other objects depend on it
DETAIL: constraint AttributeImage_id_fkey on table "AttributeImage" depends on index "Attribute_pkey"
HINT: Use DROP ... CASCADE to drop the dependent objects too.
   0: sql_migration_connector::sql_database_step_applier::apply_migration
             at migration-engine\connectors\sql-migration-connector\src\sql_database_step_applier.rs:11
   1: migration_core::state::SchemaPush
             at migration-engine\core\src\state.rs:274
```

How would I replace the table when it has dependent objects?

**Comment by pantharshit00** [comment]
Hello @iodev, our apologies for the delay in getting back to you! Due to the large number of questions in our community channels, unfortunately this slipped through the cracks. It's very important for us to address your question, so let us know in a comment if this is still an open discussion. 

If you are still facing this, can you please share your migrations folder and your schema. It is hard to infer what changes you have made from the stacktrace.

**Reply by vladi-stevanovic** [reply]
👋 Hello @iodev I'm going through pending GitHub Discussions to check if you need any help. Is this issue still open?
Please let us know if we can investigate it further or if you've already solved it, we'd love to know how!

It would be super useful for other users facing this problem! 🙌

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
